framework/src/org/checkerframework/framework/type/SupertypeFinder.java

```diff
@@ -215,7 +217,15 @@
                 AnnotatedDeclaredType dt =
                         (AnnotatedDeclaredType) atypeFactory.toAnnotatedType(superClass, false);
                 supertypes.add(dt);
-
+                // CZC: apply defaulting to the type arguments if the rawTypeAttributes of this typeElement is empty
+                if (((ClassSymbol) typeElement).getRawTypeAttributes().isEmpty()){
```

Comment:
What if there are two type arguments, but the RawTypeAttributes only contains a single type annotation.
Then the resulting type will be annotated only partially.
What happens if you remove this if?